dev/NavigationView/NavigationView.cpp

```diff
@@ -2197,6 +2177,90 @@ void NavigationView::HandleKeyEventForNavigationViewItem(const winrt::Navigation
             args.Handled(true);
             KeyboardFocusLastItemFromItem(nvi);
             break;
+        case winrt::VirtualKey::Down:
+            KeyboardFocusNextDownItem(nvi, args);
+            break;
+        case winrt::VirtualKey::Up:
+            KeyboardFocusNextUpItem(nvi, args);
+            break;
+        }
+    }
+}
+
+void NavigationView::KeyboardFocusNextUpItem(const winrt::NavigationViewItem& nvi, const winrt::KeyRoutedEventArgs& args)
+{
+    if (args.OriginalSource() == nvi)
```

Comment:
>if (args.OriginalSource() == nvi) [](start = 4, length = 33)

Nitpick: I think it's easier to read code like this by saying:

if (args.Originalsource() != nvi)
   return;

That way the rest of the function loses one level of indenting.